src/main/java/com/aws/greengrass/telemetry/impl/config/TelemetryConfig.java

```diff
@@ -117,7 +120,7 @@ public synchronized void reconfigure(LogConfigUpdate logConfigUpdate, Path store
         closeContext();
         //Reconfigure all the telemetry loggers to use the store at new path.
         for (Logger logger : context.getLoggerList()) {
-            if (!logger.getName().equals("ROOT")) {
+            if (!logger.getName().equals("ROOT") && telemetryLoggerNamesSet.contains(logger.getName())) {
```

Comment:
would it work to check that the logger contains the appender that we expect (or indeed _any_ appender)?